bakery/checkers/checkers_test.go

```diff
@@ -2,33 +2,36 @@ package checkers_test
 
 import (
 	"fmt"
-	"net"
 	"time"
 
 	jc "github.com/juju/testing/checkers"
+	"golang.org/x/net/context"
 	gc "gopkg.in/check.v1"
 	"gopkg.in/errgo.v1"
 	"gopkg.in/macaroon.v2-unstable"
 
-	"gopkg.in/macaroon-bakery.v2-unstable/bakery"
 	"gopkg.in/macaroon-bakery.v2-unstable/bakery/checkers"
 )
 
 type CheckersSuite struct{}
 
 var _ = gc.Suite(&CheckersSuite{})
 
-// Freeze time for the tests.
-var now = func() time.Time {
-	now, err := time.Parse(time.RFC3339Nano, "2006-01-02T15:04:05.123Z")
-	if err != nil {
-		panic(err)
-	}
-	*checkers.TimeNow = func() time.Time {
-		return now
-	}
+// A frozen time for the tests.
+var now = time.Date(2006, time.January, 2, 15, 4, 5, int(123*time.Millisecond), time.UTC)
+
+// testClock is a Clock implementation that always returns the above time.
+type testClock struct{}
+
+func (testClock) Now() time.Time {
 	return now
-}()
+}
+
+// FirstPartyCaveatChecker is declared here so we can avoid a cyclic
```

Comment:
The dependency on bakery was annoying when refactoring because I couldn't run the checker tests without bakery compiling. It's always nicer to avoid cyclic dependencies even in tests when reasonable.